src/dom/rect.js

```diff
@@ -361,12 +363,19 @@ export default class Rect {
 		// If there's no client rects for the Range, use parent container's bounding rect
 		// instead and adjust rect's width to simulate the actual geometry of such range.
 		// https://github.com/ckeditor/ckeditor5-utils/issues/153
+		// https://github.com/ckeditor/ckeditor5-ui/issues/317
 		else {
-			const startContainerRect = new Rect( range.startContainer.getBoundingClientRect() );
-			startContainerRect.right = startContainerRect.left;
-			startContainerRect.width = 0;
+			let startContainer = range.startContainer;
+
+			if ( startContainer.nodeType === Node.TEXT_NODE ) {
```

Comment:
Would be good to have an `isText()` util for that.

Reply:
Actually, we have it in the DomConverter already :D And we use similar logic in `scroll.js`. So at least `scroll.js` and now `rect.js` should use it. DomConverter not necessarily.

Reply:
Can I make the above change as a follow-up?

Reply:
👍